signer/signer.go

```diff
@@ -17,14 +19,17 @@ type SignerInfo struct {
 	SignatureAlgorithm SignatureAlgorithm
 	CertificateChain   []*x509.Certificate
 	Signature          []byte
+	SigningScheme      SigningScheme
```

Comment:
Nit, I'd be preferable to organize these as signed and unsigned and have some comments.

```
// SignerInfo represents a parsed signature envelope that is agnostic to signature envelope format.
type SignerInfo struct {
     
       Payload            []byte

        // Signed attributes
        SignatureAlgorithm SignatureAlgorithm
	SigningScheme      SigningScheme
	PayloadContentType PayloadContentType
	SignedAttributes   SignedAttributes

       // The signature is generated over payload and signed attributes
       Signature          []byte

       // Unsigned attributes
       CertificateChain   []*x509.Certificate
	TimestampSignature []byte
	UnsignedAttributes UnsignedAttributes
	SignatureAlgorithm SignatureAlgorithm
}
```